tests/scipy_stats_test.py

```diff
@@ -687,6 +688,174 @@ def testMultivariateNormalLogpdfBatch(self, ndim, nbatch, dtype):
     result2 = jax.vmap(lsp_stats.multivariate_normal.logpdf)(x, mean, cov)
     self.assertArraysEqual(result1, result2, check_dtypes=False)
 
+  @parameterized.named_parameters(jtu.cases_from_list(
+      {"testcase_name":
+        "_inshape={}_outsize={}_weights={}_method={}_func={}".format(
+          jtu.format_shape_dtype_string(inshape, dtype),
+          outsize, weights, method, func),
+       "dtype": dtype,
+       "inshape": inshape,
+       "outsize": outsize,
+       "weights": weights,
+       "method": method,
+       "func": func}
+      for inshape in [(50,), (3, 50), (2, 12)]
+      for dtype in jtu.dtypes.floating
+      for outsize in [None, 10]
+      for weights in [False, True]
+      for method in [None, "scott", "silverman", 1.5, "callable"]
+      for func in [None, "evaluate", "logpdf", "pdf"]))
+  def testKde(self, inshape, dtype, outsize, weights, method, func):
+    if method == "callable":
+        method = lambda kde: jax.numpy.power(kde.neff, -1./(kde.d+4))
+
+    def scipy_fun(dataset, points, w):
+      w = np.abs(w) if weights else None
+      kde = osp_stats.gaussian_kde(dataset, bw_method=method, weights=w)
+      if func is None:
+        result = kde(points)
+      else:
+        result = getattr(kde, func)(points)
+      # Note: the scipy implementation _always_ returns float64
+      return result.astype(dtype)
+
+    def lax_fun(dataset, points, w):
+      w = jax.numpy.abs(w) if weights else None
+      kde = lsp_stats.gaussian_kde(dataset, bw_method=method, weights=w)
+      if func is None:
+        result = kde(points)
+      else:
+        result = getattr(kde, func)(points)
+      return result
+
+    if outsize is None:
+      outshape = inshape
+    else:
+      outshape = inshape[:-1] + (outsize,)
+    rng = jtu.rand_default(self.rng())
+    args_maker = lambda: [
+      rng(inshape, dtype), rng(outshape, dtype), rng(inshape[-1:], dtype)]
+    self._CheckAgainstNumpy(scipy_fun, lax_fun, args_maker,
+                            tol={np.float32: 1e-3, np.float64: 1e-14})
+    self._CompileAndCheck(
+        lax_fun, args_maker, rtol={np.float32: 3e-07, np.float64: 4e-15})
+
+  @parameterized.named_parameters(jtu.cases_from_list(
+      {"testcase_name": jtu.format_test_name_suffix("", [shape], [dtype]),
+       "dtype": dtype,
+       "shape": shape}
+      for shape in [(15,), (3, 15), (1, 12)]
+      for dtype in jtu.dtypes.floating))
+  def testKdeIntegrateGaussian(self, shape, dtype):
+    @jax.numpy_dtype_promotion("standard")
```

Comment:
This decorator shouldn't be necessary if the inputs are numpy arrays (which I believe they should be, given how this function is called)

Reply:
I could have sworn that I was getting a promotion error before, but it looks like you're right.

Reply:
If you were calling `jnp.abs` instead of `np.abs` (or something similar) on inputs, then I'd believe this error could come up